### PR TITLE
Reuse code in a CURL adapter (PR for Victor's branch)

### PR DESCRIPTION
**Pull request by sergey-shandar**

### CODE REVIEW

File: sdk/core/core/inc/az_curl_slist.h [hunk 1]
```diff
@@ -11,6 +11,12 @@
 
 #include <_az_cfg_prefix.h>
 
+/**
+ * Appends a zero-terminated string to the given CURL list.
+ */
+AZ_NODISCARD az_result
+az_curl_slist_append(struct curl_slist ** const self, char const * const str);
```
Comment (vhvb1989) [inline sdk/core/core/inc/az_curl_slist.h]: why do we add this call on top of curl_easy call?

Comment (sergey-shandar) [reply sdk/core/core/inc/az_curl_slist.h]: @vhvb1989 could you update your branch from master?

Comment (vhvb1989) [reply sdk/core/core/inc/az_curl_slist.h]: done

Comment (sergey-shandar) [reply sdk/core/core/inc/az_curl_slist.h]: There are two reasons:
1. the signature of this function is compatible with actions.
2. it handles CURL errors and returns AZ_RESULT errors.

**Pull request merged**